lib/jekyll-paginate/pagination.rb

```diff
@@ -80,6 +77,18 @@ def self.template_page(site)
         end.first
       end
 
+      private
+      def build_paginated_page(site, page, pager)
+        page = Page.new(site, site.source, source_to_file_path(page), page.name)
+        page.data.delete('permalink')
+        page.pager = pager
+        page.dir = Pager.paginate_path(site, pager.page)
+        page
+      end
+      def source_to_file_path(page)
+        page.url_placeholders[:path] # meant to use page.instance_variable_get('@dir')
```

Comment:
Tried with page.path but test was failed since this includes filename.
File.dirname(page.path) seems to be another solution clearer than page.url_placeholders[:path]